### ISSUE TITLE
Stop using pickles

### ISSUE THREAD

**Issue by KOLANICH** [euske/pdfminer#205]
Pickle is insecure (in the sense that a one shouldn't unpickle pickles from untrusted machines) and may be abused for hiding malicious code. I guess that they at least should not be shipped but should be generated on user's machine. But the better solution may be to eliminate them at all.

**Comment by SnoopyLane** [comment]
Are you referring to pickled data in the pdfminer/cmap directory?  That's basically a set of data constants that is part of the code.  If strangers can modify files in your site-packages directory (or wherever you install pdfminer) then they can modify the code as well.  I don't see how pdfminer using pickles _in this way_ exposes a security risk.

If you're referring to some other use of pickles then please explain.

**Comment by KOLANICH** [comment]
>If strangers can modify files in your site-packages directory (or wherever you install pdfminer) then they can modify the code as well.

Yes,  but there is lot of `but`s. I'ts an ideal place to hide a backdoor which cannot be found by just diffing and grepping. They can also be used to place a backdoor by a compromised python installation stealthy. They also have some compatibility issues. If you wanna keep the pickles, they should be created on local machine of the one installing pdfminer.

**Comment by SnoopyLane** [comment]
Certainly your points about inspectability and compatibility are reasonable.  Personally I'm not fond of the pickles anyhow.  Not sure why these aren't just .py files, letting them compile to .pyc on first use.  But that would need removal of $CMAP_PATH otherwise it's another vector for code injection.
